### PR TITLE
Upgrade dependency versions to fix audit scanning

### PR DESCRIPTION
**Pull request by BenjiTheC**
# Description
This PR upgrades packages version to address security scanning issues from `rush-pnpm audit`. It fixes all the issues but one:

```
┌─────────────────────┬────────────────────────────────────────────────────────┐
│ moderate            │ Regular Expression Denial of Service (ReDoS) in lodash │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Package             │ lodash.trimend                                         │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Vulnerable versions │ <=4.5.1                                                │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Patched versions    │ <0.0.0                                                 │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Paths               │ ../../packages/services/installer > inquirer-s3@1.0.1  │
│                     │ > s3-ls@2.0.1 > lodash.trimend@4.5.1                   │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ More info           │ https://github.com/advisories/GHSA-29mw-wpgm-hmr9      │
└─────────────────────┴────────────────────────────────────────────────────────┘
1 vulnerabilities found
Severity: 1 moderate
```

The package `lodash.trimend@4.5.1` is the latest version of package and has not been upgraded since. In my opinion, it's not a security threat because of the way (and the frequency) we use this package in our code base.

### Packages with major version upgrade:

Version upgrades are done via running `rush upgrade-interactive --make-consistent`, when a major version is available, rush will present only major version upgrade option for the package, investigation has been done to make sure there is no breaking change introduced.

- [`nodemon`](https://www.npmjs.com/package/nodemon?activeTab=versions): `2.0.20` -> `3.1.0`
  - Breaking change (from 3.0.0): official support for node@8 dropped.
- [`luxon`](https://www.npmjs.com/package/luxon?activeTab=versions): `2.3.0` -> `3.4.4`
  - Breaking change: https://moment.github.io/luxon/#/upgrading?id=_2x-to-30, this doesn't impact the usage of this package in CDF (see [code search link](https://github.com/search?q=repo%3Aaws%2Faws-connected-device-framework%20luxon&type=code))
- [`busboy`](https://www.npmjs.com/package/busboy?activeTab=readme): `0.3.1` -> `1.6.0`
  - Breacking change: https://github.com/mscdex/busboy/issues/266. This package seems to be a legacy dependency that was somehow kept when moving from `command` package to `command-and-control` package. There is no usage of this package in command-and-control package code. (see [code search link](https://github.com/search?q=repo%3Aaws%2Faws-connected-device-framework+busboy&type=code))
- [`@types/busboy`](https://www.npmjs.com/package/@types/busboy): `0.2.3` -> `1.5.4`
  - This is the complementary major version upgrade  for busboy above

## Type of change

- [ ] *Bug fix (non-breaking change which fixes an issue)*
- [ ] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [ ] *Refactor* (existing code being refactored)
- [ ] *This change includes a documentation update*

# Submission Checklist

- [x] CI dry-run passing
- [x] Integration tests passing locally
- [ ] Change logs generated 
  - N/A

### CODE REVIEW

**Review by pcozzi — COMMENTED** [review]
Can you run integration tests on this change?

File: source/packages/services/commands/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "clean": "npx shx rm -rf dist tsconfig.tsbuildinfo bundle.zip .rush .nyc_output *.log",
     "lint": "npx eslint . --ext '.ts'",
-    "build": "npx tsc -b && npx shx cp -R 'src/config' 'dist/'",
+    "build": "npx tsc -p tsconfig.json && npx shx cp -R 'src/config' 'dist/'",
```
Comment (ts-amz) [inline source/packages/services/commands/package.json:9]: Can you add a note about why this change exists?

File: source/packages/services/commands/src/commands/commands.controller.ts [hunk 2]
```diff
@@ -132,17 +132,29 @@ export class CommandsController implements interfaces.Controller {
...
         let hasError: string;
 
         await new Promise((resolve, reject) => {
-            busboy.on('file', async (_fieldName, stream, _filename, _encoding, _mimeType) => {
-                stream.pipe(fs.createWriteStream(saveTo));
-                stream.on('limit', () => {
-                    hasError = 'filesSizeLimit';
-                });
-            });
+            busboy.on(
```
Comment (ts-amz) [inline source/packages/services/commands/src/commands/commands.controller.ts:140]: Can you add a comment explaining what's in these bigger changes so it's clear how this maintains existing functionality?

File: source/packages/services/greengrass2-provisioning/package.json [hunk 3]
```diff
@@ -36,20 +36,20 @@
...
-    "@aws-sdk/client-lambda": "3.38.0",
-    "@aws-sdk/client-s3": "3.38.0",
-    "@aws-sdk/client-sqs": "3.38.0",
-    "@aws-sdk/lib-dynamodb": "3.38.0",
-    "@aws-sdk/lib-storage": "3.38.0",
-    "@aws-sdk/signature-v4-crt": "3.38.0",
-    "@aws-sdk/smithy-client": "3.38.0",
-    "@aws-sdk/types": "3.38.0",
-    "@aws-sdk/util-dynamodb": "3.38.0",
+    "@aws-sdk/abort-controller": "3.374.0",
```
Comment (ts-amz) [inline source/packages/services/greengrass2-provisioning/package.json:39]: Why is this dependency downgraded?

File: source/packages/services/greengrass2-provisioning/src/templates/templates.models.ts [hunk 4]
```diff
@@ -114,6 +116,5 @@ export interface DeploymentPolicy {
     };
 }
 
-export type AbortConfigFailureType = 'FAILED' | 'REJECTED' | 'TIMED_OUT' | 'ALL';
```
Comment (ts-amz) [inline source/packages/services/greengrass2-provisioning/src/templates/templates.models.ts:117]: Am I reading this right that we're now getting this type from their library instead of defining it ourselves? Does the same not apply to the other exported types below? If so, can you open an issue to update them to use the library values instead of defining them in our code?

File: source/packages/services/commands/src/commands/commands.controller.ts [hunk 5]
```diff
@@ -132,17 +132,29 @@ export class CommandsController implements interfaces.Controller {
...
+                    _filename: any,
+                    _encoding: any,
+                    _mimeType: any
+                ) => {
+                    stream.pipe(fs.createWriteStream(saveTo));
+                    stream.on('limit', () => {
+                        hasError = 'filesSizeLimit';
+                    });
+                }
+            );
```
Comment (BenjiTheC) [inline source/packages/services/commands/src/commands/commands.controller.ts:157]: Even with the `suppressImplicityAnyIndexErrors: true` set up in the [tsconfig.base.json](https://github.com/aws/aws-connected-device-framework/blob/main/source/tsconfig.base.json#L25). The build still fails with implicit any error.

Adding explicit any types here to make the build pass. It's a shallow fix, but
1. This package is deprecated
2. This specific endpoint is removed in its successor module, Command And Control

File: source/packages/services/installer/package.json [hunk 6]
```diff
@@ -16,18 +16,18 @@
...
-    "@aws-sdk/client-pricing": "~3.53.0",
-    "@aws-sdk/client-s3": "~3.53.0",
-    "@aws-sdk/client-sts": "~3.53.0",
-    "@aws-sdk/client-ssm": "~3.53.0",
-    "@aws-sdk/lib-storage": "~3.53.0",
+    "@aws-sdk/client-cloudformation": "~3.575.0",
+    "@aws-sdk/client-iam": "~3.575.0",
+    "@aws-sdk/client-iot": "3.575.0",
+    "@aws-sdk/client-kms": "~3.575.0",
+    "@aws-sdk/client-lambda": "3.575.0",
```
Comment (ts-amz) [inline source/packages/services/installer/package.json:23]: Some of these updates lost the ~. Is that intentional?

Comment (BenjiTheC) [reply source/packages/services/installer/package.json:23]: This is done automatically by running `rush upgrade-interactive --make-consistent`

File: source/packages/services/commands/src/commands/commands.controller.ts (hunk 2 above)
Comment (BenjiTheC) [reply source/packages/services/commands/src/commands/commands.controller.ts:140]: See https://github.com/aws/aws-connected-device-framework/pull/210/files/3ea5837443cd6221cdfa10f4c8b47bddc7719564#r1601914368 for explanation

File: source/packages/services/greengrass2-provisioning/package.json (hunk 3 above)
Comment (BenjiTheC) [reply source/packages/services/greengrass2-provisioning/package.json:39]: Looks like this specific package's latest version is in fact `3.374.0`. See their [npm package page](https://www.npmjs.com/package/@aws-sdk/abort-controller?activeTab=versions). And this package is marked as "deprecated" with the message `This package has moved to @smithy/abort-controller`.

File: source/packages/services/commands/package.json (hunk 1 above)
Comment (BenjiTheC) [reply source/packages/services/commands/package.json:9]: This is an attempt to make `command` package pass the build. The rest of the packages use `npx tsc -p tsconfig.json` for build, so I kept this change in the PR.

File: source/packages/services/greengrass2-provisioning/src/templates/templates.models.ts (hunk 4 above)
Comment (BenjiTheC) [reply source/packages/services/greengrass2-provisioning/src/templates/templates.models.ts:117]: I believe the newer version of the dependent AWS SDK package updated their typing from a general `string` to more strict `enum`, which results in this incompatibility. And you are right, this can apply to other types defined in the code base. I have created https://github.com/aws/aws-connected-device-framework/issues/211 to track this

**Comment by BenjiTheC** [comment]
> Can you run integration tests on this change?

Integration tests are passing

**Review by pcozzi — APPROVED** [review]

**Review by ts-amz — APPROVED** [review]
